confgenerator/logging_receivers.go

```diff
@@ -249,6 +249,61 @@ func init() {
 	LoggingReceiverTypes.RegisterType(func() Component { return &LoggingReceiverTCP{} })
 }
 
+// A LoggingReceiverFluentForward represents the configuration for a Forward Protocol receiver.
+type LoggingReceiverFluentForward struct {
+	ConfigComponent `yaml:",inline"`
+
+	ListenHost string `yaml:"listen_host,omitempty" validate:"omitempty,ip"`
+	ListenPort uint16 `yaml:"listen_port,omitempty"`
+}
+
+func (r LoggingReceiverFluentForward) Type() string {
+	return "fluent_forward"
+}
+
+func (r LoggingReceiverFluentForward) Components(tag string) []fluentbit.Component {
+	if r.ListenHost == "" {
+		r.ListenHost = "127.0.0.1"
+	}
+	if r.ListenPort == 0 {
+		r.ListenPort = 24224
+	}
+
+	return []fluentbit.Component{{
+		Kind: "INPUT",
+		Config: map[string]string{
+			// https://docs.fluentbit.io/manual/pipeline/inputs/forward
+			"Name":       "forward",
+			"Tag_Prefix": tag + ".",
+			"Listen":     r.ListenHost,
+			"Port":       fmt.Sprintf("%d", r.ListenPort),
+			// https://docs.fluentbit.io/manual/administration/buffering-and-storage#input-section-configuration
+			// Buffer in disk to improve reliability.
+			"storage.type": "filesystem",
+
+			// https://docs.fluentbit.io/manual/administration/backpressure#mem_buf_limit
+			// This controls how much data the input plugin can hold in memory once the data is ingested into the core.
+			// This is used to deal with backpressure scenarios (e.g: cannot flush data for some reason).
+			// When the input plugin hits "mem_buf_limit", because we have enabled filesystem storage type, mem_buf_limit acts
+			// as a hint to set "how much data can be up in memory", once the limit is reached it continues writing to disk.
+			"Mem_Buf_Limit": "10M",
+		},
+	},
+		{
+			Kind: "FILTER",
+			Config: map[string]string{
+				"Name":   "lua",
+				"Match":  tag + ".*",
```

Comment:
Per discussion offline, we still wanna stick to the "LogEntry.logName: {RECEIVER_ID}.{ORIGINAL_TAG}" user experience.

There is some concern about the `{PIPELINE_ID}.{RECEIVER_ID}.*` match being too loose for the LUA filter here, and could potentially conflict with a `pipeline_id` that has `.` in it.

It is not a huge concern because `pipeline_id` and `receiver_id` need to be unique today. The chance of the corner case above is very low

AI:
* Double check whether the `.` between `pipeline_id` and `receiver_id` is an exact match or an `any character` match. (I thought it's an exact match, which made it less concerning)
* Double check whether a pipeline / receiver id allows `.` in them (My guess is yes. If no, then this whole problem might go away)
* Consider making the tag routing more specific for the forward plugin (e.g. We can use `fluent_forward.{PIPELINE_ID}.{RECEIVER_ID}` as the prefix instead, since the underlying routing is transparent to users.